scripts/control_pane.js

```diff
@@ -0,0 +1,310 @@
+        // translate by half difference in height and width.
+        let xOffset = 0, yOffset = 0;
+        if(sin != 0) {
+            let offset = (imageElement.height() - imageElement.width()) / 2;
+            if(offset > 0) {
+                xOffset = offset;
+            } else {
+                yOffset = -offset;
+            }
+        }
+        imageElement.css({transform: `matrix(${cosine}, ${sin}, ${-sin}, ${cosine}, ${xOffset}, ${yOffset})`});
+    }
+
+    function zoomSave() {
+        // have to recalculate transform after changing size.
+        transform();
+        localStorage.setItem(imagePercentID, percent);
+    }
+
+    percentInput.change(function() {
+        percent = this.value;
+        setZoom();
+        zoomSave();
+    });
+
+    function setPercent() {
+        // assume 100% means 1000px wide
+        let width = imageElement.width();
+        // reset width and height so that fitting does not persist
+        imageElement.width(width);
+        imageElement.height("auto");
+        percent = Math.round(width / 10);
+        percentInput.val(percent);
+        zoomSave();
+    }
+
+    let fitWidth = $("<input>", {type: 'button', value: '↔'}).click(function () {
+        imageElement.width('100%');
+        imageElement.height("auto");
+        setPercent();
+    });
+
+    let fitHeight = $("<input>", {type: 'button', value: '↕'}).click(function () {
+        imageElement.height('100%');
+        imageElement.width("auto");
+        setPercent();
+    });
+
+
+    let clockRotateInput = $("<input>", {type: 'button', value: '↻'}).click( function () {
+        let temp = sin;
+        sin = cosine;
+        cosine = -temp;
+        transform();
+    });
+
+    let anticlockRotateInput = $("<input>", {type: 'button', value: '↺'}).click( function () {
+        let temp = sin;
+        sin = -cosine;
+        cosine = temp;
+        transform();
+    });
+
+    setZoom();
+    return [
+        $("<span>", {class: "nowrap"}).append(percentInput, "% "),
+        fitWidth,
+        fitHeight,
+        clockRotateInput,
+        anticlockRotateInput
+    ];
+};
+
+function makeControlPane() {
+    // these define the position of the control bar and controls in it
+    let point = "N";
+    let begMidEnd = "M";
+
+    let contentPane = $("<div>", {class: 'overflow-auto'}).css({flex: 'auto'});
+    let container = $("<div>").css({display: 'flex', height: "100%"});
+    container.append(contentPane);
+
+    let controlPane = $("<div>", {class: 'page-interface control-panel'});
+    // control2 contains the controls, control1 & control3 adjust the layout
+    let control1 = $("<div>");
+    let control2 = $("<div>").css({flex: '0 1 auto'});
+    let control3 = $("<div>");
+    controlPane.append(control1, control2, control3);
+
+    let menu = $("<div>", {class: "nav-menu"});
+
+    let menuButton = $("<input>", {type: 'button', value: '⌘'}).click(function () {
```

Comment:
I had no clue what this character meant. I don't have a mac and am not used to seeing it. Unicode calls it "PLACE OF INTEREST SIGN" but using the sandbox, I had no idea what to expect when clicking it. Same for the buttons about sizing and moving the controls. Could we have some alt text at least?

Reply:
I think of it as a cloverleaf, but it's most familiar to Mac users as the label for the command key (where Windows uses `ctrl-c`/`v` for copy and paste, Macs use `cmd-c`/`v` for copy and paste, as an example).

I did a half-hearted search of Unicode characters to see if I could find a unicode for a `+` with up/down/left/right arrows on it like Casey's mockup has, but I didn't find anything (there are so many versions of arrows that you'd think there would be, but I didn't find one). Ditto for a compass rose. There is a compass, but I'm not sure that's a good choice. Maybe a diamond? How would something like `❖` or `⟡` or `⟐` work?

Alt text is a good idea.

Reply:
I think using ⌘ is actually rather brilliant, as it _is_ a place of interest. Much of this interface is going to be "try it and see what happens". Agreed that alt text will really help for discovery.

Reply:
Yes, that would be good.